File: elecciones-main/src/main/java/com/eleccciones/cliente/data/entities/security/Usuario.java
package com.eleccciones.cliente.data.entities.security;

import com.eleccciones.cliente.data.entities.candidatos.Contrato;
import com.eleccciones.cliente.data.entities.candidatos.Entidad;
import com.eleccciones.cliente.data.entities.core.impl.AbstractEntity;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.io.Serializable;


@Entity
@Table(name = "usuario", uniqueConstraints = {@UniqueConstraint(columnNames = {"login"})})
@Getter
@Setter
public class Usuario extends AbstractEntity implements Serializable {
    private static final long serialVersionUID = -7310305185273678943L;

    @NotEmpty
    @Column(name = "login", nullable = false)
    @Size(min = 10, max = 25, message = "{entity.auditoria.username}")
    private String username;

    @NotEmpty
    @JsonIgnore
    @Column(name = "clave", nullable = false)
    private String password;

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_entidad", referencedColumnName = "id")
    private Entidad entidad;

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_contrato", referencedColumnName = "id")
    private Contrato contrato;

    @NotNull
    @Column(name = "activo")
    private String activo;

    @Column(name = "habilitado_movil")
    private Integer habilitadoMovil;

    @Column(name = "sesion_movil")
    private Integer sessionMovil;

    @Column(name = "mac_movil")
    private String macMovil;

    @Column(name = "habilitado_web")
    private Integer habilitadoWeb;

    @Column(name = "sesion_web")
    private Integer sessionWeb;
    
    @Override
    public String toString() {
        return "Usuario [" + (username != null ? "username=" + username + ", " : "")
                + (password != null ? "password=" + password : "") + "]";
    }
}

File: elecciones_app_movil/lib/data/model/auth/usuario.dart
import 'package:elecciones_app_movil/data/model/auth/contrato.dart';
import 'package:elecciones_app_movil/data/model/auth/entidad.dart';

class Usuario {
  Usuario({
    this.id,
    this.username,
    this.entidad,
    this.contrato,
    this.activo,
    this.habilitadoMovil,
    this.sessionMovil,
    this.macMovil,
    this.habilitadoWeb,
    this.sessionWeb,
    this.roles,
  });

  int? id;
  String? username;
  Entidad? entidad;
  Contrato? contrato;
  String? activo;
  String? habilitadoMovil;
  int? sessionMovil;
  String? macMovil;
  String? habilitadoWeb;
  int? sessionWeb;
  List<String>? roles;

  Usuario copyWith({
    int? id,
    String? username,
    Entidad? entidad,
    Contrato? contrato,
    String? activo,
    String? habilitadoMovil,
    int? sessionMovil,
    String? macMovil,
    String? habilitadoWeb,
    int? sessionWeb,
    List<String>? roles,
  }) =>
      Usuario(
        id: id ?? this.id,
        username: username ?? this.username,
        entidad: entidad ?? this.entidad,
        contrato: contrato ?? this.contrato,
        activo: activo ?? this.activo,
        habilitadoMovil: habilitadoMovil ?? this.habilitadoMovil,
        sessionMovil: sessionMovil ?? this.sessionMovil,
        macMovil: macMovil ?? this.macMovil,
        habilitadoWeb: habilitadoWeb ?? this.habilitadoWeb,
        sessionWeb: sessionWeb ?? this.sessionWeb,
        roles: roles ?? this.roles,
      );

  @override
  String toString() {
    return 'Usuario{id: $id, username: $username, entidad: $entidad, contrato: $contrato, activo: $activo, habilitadoMovil: $habilitadoMovil, sessionMovil: $sessionMovil, macMovil: $macMovil, habilitadoWeb: $habilitadoWeb, sessionWeb: $sessionWeb, roles: $roles}';
  }

  factory Usuario.fromJson(Map<String, dynamic> json) => Usuario(
        id: json["id"],
        username: json["username"],
        entidad: Entidad.fromJson(json["entidad"]),
        contrato: Contrato.fromJson(json["contrato"]),
        activo: json["activo"],
        habilitadoMovil: json["habilitadoMovil"],
        sessionMovil: json["sessionMovil"],
        macMovil: json["macMovil"],
        habilitadoWeb: json["habilitadoWeb"],
        sessionWeb: json["sessionWeb"],
        roles: List<String>.from(json["roles"].map((x) => x)),
      );

  Map<String, dynamic> toJson() => {
        "id": id,
        "username": username,
        "entidad": entidad!.toJson(),
        "contrato": contrato!.toJson(),
        "activo": activo,
        "habilitadoMovil": habilitadoMovil,
        "sessionMovil": sessionMovil,
        "macMovil": macMovil,
        "habilitadoWeb": habilitadoWeb,
        "sessionWeb": sessionWeb,
        "roles": List<dynamic>.from(roles!.map((x) => x)),
      };
}
